account_fiscal_position_vies_warning/models/account_move.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2025 Moduon Team S.L. <info@moduon.team>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_post(self):
+        if not self.env.company.vat_check_vies:
+            return super().action_post()
+        for record in self:
+            if (
+                record.fiscal_position_id
+                in self.env.company.check_vies_fiscal_position_ids
+                and not record.partner_id.vies_valid
+            ):
+                raise ValidationError(
+                    _(
+                        "You can't set the Fiscal Position to %(fiscal_pos)s, "
+                        "%(partner_name)s doesn't pass VIES VAT number validation!"
+                    )
+                    % {
+                        "fiscal_pos": record.fiscal_position_id.name,
+                        "partner_name": self.partner_id.name,
+                    }
+                )
```

Comment:
A method at partner's level for raising this message would avoid the need for translating it twice